### PR TITLE
Feature/entity apis

### PR DESCRIPTION
**Pull request by borkarsaish65**

### CODE REVIEW

**Review by VISHNUDAS-tunerlabs — CHANGES_REQUESTED** [review]
Reviewed on 28 oct 2024

File: controllers/v1/observationsController.js [hunk 1]
```diff
@@ -627,6 +628,22 @@ module.exports = class Observations extends Abstract {
     });
   }
 
+  async targetedEntity(req) {
```
[the remark this answers is not in the document]
Comment (borkarsaish65) [reply controllers/v1/observationsController.js:631]: done

File: generics/services/entity-management.js [hunk 2]
```diff
@@ -20,21 +20,28 @@ const validateEntity = process.env.VALIDATE_ENTITIES;
  */
 
 // Function to find entity documents based on the given filter and projection
-const entityDocuments = function (filterData = 'all', projection = 'all') {
+const entityDocuments = function (filterData = 'all', projection = 'all',page=null,limit=null) {
```
[the remark this answers is not in the document]
Comment (borkarsaish65) [reply generics/services/entity-management.js:23]: done

File: generics/services/entity-management.js [hunk 3]
```diff
@@ -218,9 +224,114 @@ const listByEntityType = async function (entityTypeId,userToken,pageSize,pageNo)
...
+    try {
+      // Define the URL for the user role extension API
+      const url = entityManagementServiceUrl+messageConstants.endpoints.USER_ROLE_EXTENSION;
+
+      // Set the options for the HTTP POST request
+      const options = {
+        headers: {
+          'content-type': 'application/json',
+          'internal-access-token': process.env.INTERNAL_ACCESS_TOKEN,
+          'X-auth-token': authToken, // Use the passed authentication token
```
[the remark this answers is not in the document]
Comment (borkarsaish65) [reply generics/services/entity-management.js:250]: done

File: module/observationSubmissions/validator/v1.js [hunk 4]
```diff
@@ -37,8 +37,6 @@ module.exports = (req) => {
         .checkQuery('entityId')
         .exists()
         .withMessage('required entity id')
-        .isMongoId()
```
[the remark this answers is not in the document]
Comment (borkarsaish65) [reply module/observationSubmissions/validator/v1.js:40]: removed as we are not sure if entity management is off or some external enitty service is used it wont necessarily be mongo id. Shubham from tekdi too got this issue as they dont use mongoid for their entities

File: generics/services/entity-management.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (borkarsaish65) [reply generics/services/entity-management.js]: done

**Review by VISHNUDAS-tunerlabs — APPROVED** [review]

**Pull request merged**